Remotion/Data/DomainObjects/Persistence/Rdbms/StorageProviderCommands/Factories/QueryCommandFactory.cs

```diff
@@ -74,7 +74,7 @@ public IDataStoragePropertyDefinitionFactory DataStoragePropertyDefinitionFactor
       return new MultiObjectLoadCommand<DataContainer?>(new[] { Tuple.Create(dbCommandBuilder, dataContainerReader) });
     }
 
-    public virtual IStorageProviderCommand<IEnumerable<IQueryResultRow>, IRdbmsProviderCommandExecutionContext> CreateForCustomQuery (IQuery query)
+    public virtual IRdbmsProviderReadWriteCommand<IEnumerable<IQueryResultRow>> CreateForCustomQuery (IQuery query)
```

Comment:
The query would always be executed in a read-write context, because it is a read-write command.
This is wrong; it should be changed to read-only command, so that the QueryType and IStorageAccessResolver (see RM-9113) can determine whether to use the read-only or read-write context.